### ISSUE TITLE
Survey_questions not pulling statements for matrix table questions?

### ISSUE THREAD

**Issue by pschatz25** [ropensci/qualtRics#185]
Hello,

Great package! It is saving me *a lot* of time wrangling Qualtrics data.

When I run survey_questions for a survey with matrix table type questions I"m only getting the questions returned but not the statements. Is there a way to pull the statements too? 

Thanks!

**Comment by juliasilge** [comment]
So glad it is helpful so far!

Can you give me some more details on what you mean by "questions" and "statements" here? There are a lot of different moving parts that go together and often we don't use the same names for these. If you can show specific examples of what you are missing, that would be very helpful. 

Also if you want to check out two current open issues we have on matrix questions to see if they address the particular issue you have, they are #144 and #157.

**Comment by pschatz25** [comment]
I think I'm asking about something different. The fetch_surveys() function is returning results exactly as I would expect. My issue is specifically with the survey_questions() function. It's returning the one 'overall' question for a matrix table type question, but not the sub-questions/statements. 

So I have a survey with 'Q4' as a matrix table question. From fetch_surveys() I get:
`length(grep(x = colnames(mysurvey_text), pattern = "^Q4"))`
`8`
From survey_questions() I get:
`sum(grepl('^Q4', questions$qname))`
`1`

Hopefully that helps explain things. I can work around it by taking the column labels and attributes off the fetch_surveys() results.

**Comment by juliasilge** [comment]
Ah thank you; I understand what you are saying now. This is related to the underlying Qualtrics API. Here is one of my example surveys, where `QID1`/`Q1` is a matrix question. There is information about subquestions returned in the API but [right now we only map over question name and question text](https://github.com/ropensci/qualtRics/blob/ccdd4f1be2e9bad666ec73dff7f863e77ca64dea/R/survey_questions.R#L53) (as well as whether the response is forced).

It may be possible to change how we deal with the information returned in the API to also map over subquestions when they exist. If you are interested in submitting a PR for this, it would be a good feature to support!

``` r
surveys_url <- qualtRics:::create_surveys_url(Sys.getenv("QUALTRICS_BASE_URL"))
surveys_url <- paste0(surveys_url, "SV_5BJRo2RGHajIlOB")
resp <- qualtRics:::qualtrics_api_request("GET", surveys_url)
resp$result$questions$QID1
#> $questionType
#> $questionType$type
#> [1] "Matrix"
#> 
#> $questionType$selector
#> [1] "Likert"
#> 
#> $questionType$subSelector
#> [1] "SingleAnswer"
#> 
#> 
#> $questionText
#> [1] "Are you aware of the <strong>sourdough bread</strong> features listed below?"
#> 
#> $questionLabel
#> NULL
#> 
#> $validation
#> $validation$doesForceResponse
#> [1] TRUE
#> 
#> 
#> $questionName
#> [1] "Q1"
#> 
#> $subQuestions
#> $subQuestions$`1`
#> $subQuestions$`1`$recode
#> [1] "1"
#> 
#> $subQuestions$`1`$description
#> [1] "Distraction during quarantine"
#> 
#> $subQuestions$`1`$choiceText
#> [1] "Distraction during quarantine"
#> 
#> $subQuestions$`1`$variableName
#> NULL
#> 
#> $subQuestions$`1`$imageDescription
#> NULL
#> 
#> 
#> $subQuestions$`2`
#> $subQuestions$`2`$recode
#> [1] "2"
#> 
#> $subQuestions$`2`$description
#> [1] "Texture"
#> 
#> $subQuestions$`2`$choiceText
#> [1] "Texture"
#> 
#> $subQuestions$`2`$variableName
#> NULL
#> 
#> $subQuestions$`2`$imageDescription
#> NULL
#> 
#> 
#> $subQuestions$`3`
#> $subQuestions$`3`$recode
#> [1] "3"
#> 
#> $subQuestions$`3`$description
#> [1] "Flavor"
#> 
#> $subQuestions$`3`$choiceText
#> [1] "Flavor"
#> 
#> $subQuestions$`3`$variableName
#> NULL
#> 
#> $subQuestions$`3`$imageDescription
#> NULL
#> 
#> 
#> $subQuestions$`4`
#> $subQuestions$`4`$recode
#> [1] "4"
#> 
#> $subQuestions$`4`$description
#> [1] "Color"
#> 
#> $subQuestions$`4`$choiceText
#> [1] "Color"
#> 
#> $subQuestions$`4`$variableName
#> NULL
#> 
#> $subQuestions$`4`$imageDescription
#> NULL
#> 
#> 
#> $subQuestions$`5`
#> $subQuestions$`5`$recode
#> [1] "5"
#> 
#> $subQuestions$`5`$description
#> [1] "Nutritional value"
#> 
#> $subQuestions$`5`$choiceText
#> [1] "Nutritional value"
#> 
#> $subQuestions$`5`$variableName
#> NULL
#> 
#> $subQuestions$`5`$imageDescription
#> NULL
#> 
#> 
#> 
#> $choices
#> $choices$`1`
#> $choices$`1`$recode
#> [1] "1"
#> 
#> $choices$`1`$description
#> [1] "Aware of it"
#> 
#> $choices$`1`$choiceText
#> [1] "Aware of it"
#> 
#> $choices$`1`$imageDescription
#> NULL
#> 
#> $choices$`1`$analyze
#> [1] TRUE
#> 
#> 
#> $choices$`2`
#> $choices$`2`$recode
#> [1] "2"
#> 
#> $choices$`2`$description
#> [1] "Not aware of it"
#> 
#> $choices$`2`$choiceText
#> [1] "Not aware of it"
#> 
#> $choices$`2`$imageDescription
#> NULL
#> 
#> $choices$`2`$analyze
#> [1] TRUE
#> 
#> 
#> 
#> $randomization
#> $randomization$type
#> [1] "all"
```

<sup>Created on 2020-08-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

**Comment by pschatz25** [comment]
Sounds good. Thanks for the explanation. I have viable workaround for my current project so it's not a critical issue. If I have some time after this project maybe I'll take a stab at this feature.

**Comment by juliasilge** [comment]
I believe this issue is addressed and dealt with in the changes made in #199. You can now access a **column mapping** after downloading the data (it is saved as an attribute) that includes the subquestions/substatements in the `sub` column:

``` r
library(qualtRics)
res <- fetch_survey("SV_5BJxxxxxxx")
#>   |                                                                              |                                                                      |   0%  |                                                                              |=========================================================             |  82%  |                                                                              |======================================================================| 100%
#> 
#> ── Column specification ────────────────────────────────────────────────────────
#> cols(
#>   .default = col_character(),
#>   StartDate = col_datetime(format = ""),
#>   EndDate = col_datetime(format = ""),
#>   Progress = col_double(),
#>   `Duration (in seconds)` = col_double(),
#>   Finished = col_logical(),
#>   RecordedDate = col_datetime(format = ""),
#>   RecipientLastName = col_logical(),
#>   RecipientFirstName = col_logical(),
#>   RecipientEmail = col_logical(),
#>   ExternalReference = col_logical(),
#>   LocationLatitude = col_double(),
#>   LocationLongitude = col_double(),
#>   Q1007 = col_double(),
#>   Q1_DO_1 = col_double(),
#>   Q1_DO_2 = col_double(),
#>   Q1_DO_3 = col_double(),
#>   Q1_DO_4 = col_double(),
#>   Q1_DO_5 = col_double(),
#>   SolutionRevision = col_double(),
#>   FL_6_DO_FL_7 = col_double()
#>   # ... with 4 more columns
#> )
#> ℹ Use `spec()` for the full column specifications.
attr(res, "column_map")
#> # A tibble: 46 x 7
#>    qname        description      main        sub   ImportId    timeZone choiceId
#>    <chr>        <chr>            <chr>       <chr> <chr>       <chr>    <chr>   
#>  1 StartDate    Start Date       Start Date  ""    startDate   Z        <NA>    
#>  2 EndDate      End Date         End Date    ""    endDate     Z        <NA>    
#>  3 Status       Response Type    Response T… ""    status      <NA>     <NA>    
#>  4 IPAddress    IP Address       IP Address  ""    ipAddress   <NA>     <NA>    
#>  5 Progress     Progress         Progress    ""    progress    <NA>     <NA>    
#>  6 Duration (i… Duration (in se… Duration (… ""    duration    <NA>     <NA>    
#>  7 Finished     Finished         Finished    ""    finished    <NA>     <NA>    
#>  8 RecordedDate Recorded Date    Recorded D… ""    recordedDa… Z        <NA>    
#>  9 ResponseId   Response ID      Response ID ""    _recordId   <NA>     <NA>    
#> 10 RecipientLa… Recipient Last … Recipient … ""    recipientL… <NA>     <NA>    
#> # … with 36 more rows
```

<sup>Created on 2021-01-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

If you have further questions or problems, please open a new issue! 🙌
